libraries/splunk_app.rb

```diff
@@ -126,13 +174,80 @@ def create_splunk_directory(path)
         dir.group(node['splunk']['group'])
         dir.mode('0755')
         dir.run_action(:create)
-        new_resource.updated_by_last_action(dir.updated_by_last_action?)
+        new_resource.updated_by_last_action(true) if dir.updated_by_last_action?
+      end
+
+      def download_and_install # rubocop:disable PerceivedComplexity, CyclomaticComplexity
+        return if new_resource.url.nil? || new_resource.url.empty?
+
+        expected_version = CernerSplunk::AppVersion.new new_resource.version
+        installed_version = CernerSplunk::AppVersion.new @current_resource.version
+
+        return if expected_version.type && expected_version == installed_version
```

Comment:
Something else I'm a bit confused about on this line.  It looks like you have [defined ==](https://github.com/cerner/cerner_splunk/pull/48/files#diff-940247b1e7418e1207338ac1de247744R43) as only comparing AppVersion#version so if, for example, I installed "1.0 beta1" and then tried to update to "1.0 beta2", it would result in a no-op?